### PR TITLE
New client validation stride

### PR DESCRIPTION
**Pull request by sdquiring**
Only calculate the elements that will be used for validation.

Not sure why I didn't see that the old client was doing this before.  This should reduce test times and allow further reduction especially with the complex tests.

### CODE REVIEW

File: Tensile/Source/client/source/Reference.cpp [hunk 1]
```diff
@@ -84,52 +85,48 @@ namespace Tensile
...
-            for(size_t freeBNum = 0; freeBNum < freeBCount; freeBNum++)
-            {
-                std::vector<size_t> aCoord(problem.a().dimensions());
-                std::vector<size_t> bCoord(problem.b().dimensions());
-                std::vector<size_t> cCoord(problem.c().dimensions());
-                std::vector<size_t> dCoord(problem.d().dimensions());
+            auto dCount = d.totalLogicalElements();
+            //for(size_t batchNum = 0; batchNum < batchCount; batchNum++)
+            //for(size_t freeANum = 0; freeANum < freeACount; freeANum++)
+            //for(size_t freeBNum = 0; freeBNum < freeBCount; freeBNum++)
```
Comment (bensander) [inline Tensile/Source/client/source/Reference.cpp]: remove commented-out code?

Comment (sdquiring) [reply Tensile/Source/client/source/Reference.cpp]: I did, in the cleanup commit: https://github.com/ROCmSoftwarePlatform/Tensile/pull/681/files

File: Tensile/Source/client/source/ReferenceValidator.cpp [hunk 2]
```diff
@@ -80,7 +80,11 @@ namespace Tensile
             {
                 m_problem = problem;
                 m_referenceInputs = m_dataInit->prepareCPUInputs();
-                SolveCPU(problem, *m_referenceInputs);
+                m_validationStride = 1;
+                if(m_elementsToValidate > 0 && m_elementsToValidate < problem.d().totalLogicalElements())
+                    m_validationStride = NextPrime(problem.d().totalAllocatedElements() / m_elementsToValidate);
```
[the remark this answers is not in the document]
Comment (sdquiring) [reply Tensile/Source/client/source/ReferenceValidator.cpp]: I'm replicating the behaviour from the old client: https://github.com/ROCmSoftwarePlatform/Tensile/blob/develop/Tensile/Source/Client.h#L507

I can see the logic:
 - If the stride is 1, we only check the first few elements, which would hide a bug that only computed the first few elements.
 - If the stride is a multiple of any of the strides of D, we could be only checking a certain geometric region of the tensor, which could hide a bug that had some kind of geometric regularity (relative to workgroup size, or tensor stride, etc).

[the remark this answers is not in the document]
Comment (sdquiring) [reply Tensile/Source/client/source/ReferenceValidator.cpp]: Also see the comment: https://github.com/ROCmSoftwarePlatform/Tensile/blob/develop/Tensile/Common.py#L63

**Pull request merged**